gapic/schema/wrappers.py

```diff
@@ -786,6 +808,7 @@ def grpc_stub_type(self) -> str:
             server='stream' if self.server_streaming else 'unary',
         )
 
+    # TODO(yonmg): figure out why idempotent is reliant on http annotation
```

Comment:
http GET requests are automatically considered idempotent because they do not change data on the server. Post requests are usually non-idempotent, because it is usually unsafe retry write operation if it failed, because it may lead for example into writing the same information twice on the server.

Reply:
I get the GET is idempotent but couldn't there be methods without an http annotation that are also idempotent? are grpc only methods inherently not so?